src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -1476,6 +1486,7 @@ public static void GetString_BadUtf8()
 
                 Assert.Equal(JsonValueType.String, root.Type);
                 Assert.Throws<InvalidOperationException>(() => root.GetString());
+                // Assert.Throws<InvalidOperationException>(() => root.ValueEquals("throws-anyway")); // TODO: Fix error should throw but doesn't
```

Comment:
Uncomment this test, and fix before merging.

Reply:
the current GetString method throws with this message

"Cannot transcode invalid UTF-8 JSON text to UTF-16 string."

 but ValueEquals does not throw.

Reply:
Why should `ValueEquals` throw? Valid UTF-16 string compared to some invalid UTF-8 text should return false, no?

`GetString()` is throwing because it is trying to materialize UTF-8 text into UTF-16 which is where the check happens and we throw.

We should probably just remove the line, but not worth the reset.